### PR TITLE
MGDOBR-1375: Timeout [v2]: DataPlane to contain timeout property and send to ControlPlane

### PR DESCRIPTION
**Pull request by manstis**
[https://issues.redhat.com/browse/MGDOBR-1375](https://issues.redhat.com/browse/MGDOBR-1375)

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-XXX] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

File: openapi/smartevents_mgmt_v2.yaml [hunk 1]
```diff
@@ -1808,6 +1808,7 @@ components:
       required:
       - kind
       - id
+      - name
```
Comment (manstis) [inline openapi/smartevents_mgmt_v2.yaml:1811]: @r00ta This seemed to be missing from earlier (I wasn't expecting it to change in this PR!!)

File: openapi/smartevents_mgmt_v2.yaml [hunk 2]
```diff
@@ -1888,6 +1889,7 @@ components:
...
```
Comment (manstis) [inline openapi/smartevents_mgmt_v2.yaml:1892]: @r00ta This seemed to be missing from earlier (I wasn't expecting it to change in this PR!!)

File: infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/dto/BaseResourceDTO.java [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+import java.time.ZonedDateTime;
+import java.util.Objects;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.redhat.service.smartevents.infra.v2.api.models.OperationType;
+
+@JsonIgnoreProperties(ignoreUnknown = true)
+public abstract class BaseResourceDTO {
```
Comment (manstis) [inline infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/dto/BaseResourceDTO.java:11]: @r00ta A new base class that may be of interest for your `Sink` and `Sources`?

**Review by lucamolteni — APPROVED** [review]

File: infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/dto/BaseResourceDTO.java [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    private String owner;
+
+    @JsonProperty("operation")
+    private OperationType operationType;
+
+    @JsonProperty("generation")
+    private long generation;
+
+    @JsonProperty("expiry")
+    private ZonedDateTime expiry;
```
Comment (robpblake) [inline infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/dto/BaseResourceDTO.java:31]: @manstis I think it is probably better for us to communicate the timeout period in seconds and let the data plane calculate when it's deadline for deployment is. So essentially the DP is given `X` seconds to deploy something from the first time it sees the resource from the CP.

I'd also say something like `deploymentDeadline` or `readyDeadline` is a bit more explanatory as to the intent of the field here.

Comment (manstis) [reply infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/dto/BaseResourceDTO.java:31]: OK, changed to a property called `timeout` instead (being the number of seconds).

`Deadline` suggests a timestamp to me.

File: infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/dto/BaseResourceDTO.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    private String owner;
+
+    @JsonProperty("operation")
+    private OperationType operationType;
+
+    @JsonProperty("generation")
+    private long generation;
+
+    @JsonProperty("timeout")
+    private int timeout;
```
Comment (r00ta) [inline infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/dto/BaseResourceDTO.java:30]: nitpick: better to add the unit like `seconds` in the name to make the contract more self-descriptive

**Review by r00ta — APPROVED** [review]

File: infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/dto/BaseResourceDTO.java (hunk 5 above)
Comment (robpblake) [inline infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/dto/BaseResourceDTO.java:30]: @manstis Lets make this `timeoutSeconds` to match the accessors and make the unit of the timeout clear 👍

**Review by robpblake — APPROVED** [review]
Very minor comment from me @manstis , but other than that, looks good 👍

File: infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/dto/BaseResourceDTO.java (hunk 5 above)
Comment (manstis) [reply infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/dto/BaseResourceDTO.java]: @r00ta @robpblake renamed.

Comment (manstis) [reply infra-parent/infra-v2/src/main/java/com/redhat/service/smartevents/infra/v2/api/models/dto/BaseResourceDTO.java]: @r00ta @robpblake renamed.

**Comment by vaibhavjainwiz** [comment]
@manstis 
Could you please add these properties in `kustomize` scripts as well.

**Review by vaibhavjainwiz — COMMENTED** [review]
@manstis
Could you please add these properties in kustomize scripts as well.

**Comment by manstis** [comment]
@vaibhavjainwiz 

> Could you please add these properties in kustomize scripts as well.

Done.

**Review by vaibhavjainwiz — APPROVED** [review]

**Pull request merged**